hpcgap/lib/mapping.gi

```diff
@@ -222,19 +239,18 @@ InstallGlobalFunction( Images, function ( arg )
         map := arg[1];
         elm := arg[2];
 
-        if not IsGeneralMapping( map ) then
-          Error( "<map> must be a general mapping" );
-        fi;
-
         # image of a single element <elm> under the mapping <map>
-        if     FamSourceEqFamElm( FamilyObj( map ), FamilyObj( elm ) )
-           and elm in Source( map ) then
-
+        if     FamSourceEqFamElm( FamilyObj( map ), FamilyObj( elm ) ) then
+          if not elm in Source( map ) then
+            ErrorNoReturn( "<elm> must be an element of Source(<map>)" );
+          fi;
           return ImagesElm( map, elm );
 
         # image of a set or list of elments <elm> under the mapping <map>
-        elif     CollFamSourceEqFamElms( FamilyObj( map ), FamilyObj(elm) )
-             and IsSubset( Source( map ), elm ) then
+        elif CollFamSourceEqFamElms( FamilyObj( map ), FamilyObj(elm) ) then
+          if not IsSubset( Source( map ), elm ) then
+            ErrorNoReturn( "<coll> must be a subset of Source(<map>)" );
```

Comment:
Same comments as above, I won't repeat the text but my comments will apply to all the other places where you have written `coll` in the ways that I have complained about.